portality/lib/es_snapshots.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime, timedelta
+import requests
+from portality.core import app
+
+
+class BadSnapshotMetaException(Exception):
+    pass
+
+
+class TodaySnapshotMissingException(Exception):
+    pass
+
+
+class FailedSnapshotException(Exception):
+    pass
+
+
+class SnapshotDeleteException(Exception):
+    pass
+
+
+class ESSnapshot(object):
+    def __init__(self, snapshot_json):
+        self.data = snapshot_json
+        self.name = snapshot_json['snapshot']
+        self.state = snapshot_json['state']
+        self.datetime = datetime.utcfromtimestamp(snapshot_json['start_time_in_millis'] / 1000)
+
+    def __str__(self):
+        return str(self.__dict__)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def delete(self):
```

Comment:
Delete is a client operation, rather than a model object operation.  If you're going to separate the client and the models it deals with (which I think is right), then you should put the operational stuff into the client.

Reply:
Thanks, I'll change that